server/src/providers/completion.ts

```diff
@@ -0,0 +1,197 @@
+import {
+    CompletionParams,
+    CompletionItem,
+    CompletionItemKind,
+    MarkupKind,
+} from 'vscode-languageserver/node';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { getAutocompleteContextAtPosition } from '../utils/positions';
+import { trackInstanceDefinitions } from '../utils/instance';
+import {
+    functionsData,
+    classesData,
+    callbacksData,
+    extractClassConstructors,
+} from '../services/documentation-service';
+import * as fs from 'fs';
+import { ClassInfo } from '../config/types';
+import { EIDOS_CLASSES_PATH, SLIM_CLASSES_PATH } from '../config/paths';
+
```

Comment:
this is the fault of my original implementation-- we are doing duplicate file reads for class constructors. i wonder if could use the already-loaded `classesData` from the documentation service instead? what do you think? not a blocker for this PR though